### PR TITLE
Parsing orderbook changes

### PR DESCRIPTION
**Pull request by lumberj**

### CODE REVIEW

File: transactionparser/src/orderbookchanges.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ */
+function parseOrderbookChanges(metadata) {
+  var orderbookChanges = utils.normalizeNodes(metadata).map(function(node) {
+    if (node.entryType === 'Offer') {
+      return parseOrderbookChange(node);
+    }
+  });
+
+  return _.compact(orderbookChanges);
+}
```
Comment (clark800) [inline transactionparser/src/orderbookchanges.js]: You can use _.filter then map parseOrderbookChange over the filtered array.

File: transactionparser/src/orderbookchanges.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  return {
+    account: node.finalFields.Account,
+    taker_pays: parseChangeAmount(node, 'TakerPays'),
+    taker_gets: parseChangeAmount(node, 'TakerGets'),
+    sequence: node.finalFields.Sequence,
+    status: status
+  };
+
+  function parseChangeAmount(node,type) {
```
Comment (clark800) [inline transactionparser/src/orderbookchanges.js]: Did you run eslint on this yet? There's a missing space.

File: transactionparser/src/orderbookchanges.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    var previousAmount;
+    var finalAmount;
+    var changeAmount;
+    var finalValue;
+    var prevValue;
```
Comment (clark800) [inline transactionparser/src/orderbookchanges.js]: Even though javascript is going to hoist, I still think it's preferable to declare variables in the tightest possible scope (applies to all except changeAmount).

Comment (lumberj) [inline transactionparser/src/orderbookchanges.js]: ["Assign variables at the top of their scope"](https://github.com/airbnb/javascript#variables)

File: transactionparser/src/orderbookchanges.js (hunk 1 above)
Comment (lumberj) [inline transactionparser/src/orderbookchanges.js]: Will switch to _filter

File: transactionparser/src/orderbookchanges.js (hunk 3 above)
Comment (clark800) [inline transactionparser/src/orderbookchanges.js]: I don't know if we should follow that advice. I'm guessing they wrote that under the assumption that you are not using a linter. Are there any potential issues with hoisting that a linter won't catch? A linter should warn you if you ever use a variable before it is declared, so I can't think of any. Keeping it in the tightest possible scope is considered best practice in other languages, increases cohesion, and reduces lines of code. And hopefully future versions of javascript will implement block-level scoping.

Comment (lumberj) [inline transactionparser/src/orderbookchanges.js]: That's the style guide we chose to follow I think

Comment (clark800) [inline transactionparser/src/orderbookchanges.js]: Sure, but let's not treat it like gospel. If there is a good reason to justify the rule, then I'm all for it, but I would prefer not to sacrifice code quality for the sake of following a rule that was potentially made under false assumptions.

File: transactionparser/src/orderbookchanges.js (hunk 2 above)
Comment (lumberj) [inline transactionparser/src/orderbookchanges.js]: I ran eslint with the config file from ripple-lib. There are many errors. Is there a list of things we choose to ignore? The missing space is a `warning` but there are also errors such as "Missing 'use strict'"

File: transactionparser/src/orderbookchanges.js (hunk 3 above)
Comment (lumberj) [inline transactionparser/src/orderbookchanges.js]: Made the changes and updated PR

**Comment by clark800** [comment]
:+1: I like the nested functions

**Pull request merged**